webapp/src/actions/index.js

```diff
@@ -339,28 +393,42 @@ export function updateRhsState(rhsState) {
     };
 }
 
-export function openCreateIssueModal(postId) {
+export function openCreateIssueModalWithPost(postId) {
     return {
-        type: ActionTypes.OPEN_CREATE_ISSUE_MODAL,
+        type: ActionTypes.OPEN_CREATE_ISSUE_MODAL_WITH_POST,
         data: {
             postId,
         },
     };
 }
 
-export function openCreateIssueModalWithoutPost(title, channelId) {
+export function openCreateOrUpdateIssueModal(messageData) {
+    return {
+        type: ActionTypes.OPEN_CREATE_OR_UPDATE_ISSUE_MODAL,
+        data: {
+            messageData,
+        },
```

Comment:
```suggestion
        data: messageData,
```
Can't we do it like this? If yes, do this in the below actions as well.

Reply:
No, we can't do like this

Reply:
Why?

Reply:
Here we have 2 different action type which has different info. But in reducer these action types return same state. So its not good to directly initialise `data` with `messageData`